sled-agent/src/illumos/dladm.rs

```diff
@@ -221,6 +221,21 @@ impl Dladm {
         Ok(())
     }
 
+    /// Verify that the given link exists
+    pub fn verify_link(link: &str) -> Result<Link, FindPhysicalLinkError> {
```

Comment:
Mostly taken from https://github.com/oxidecomputer/omicron/pull/1820 - but I'm also returning the `Link` directly here.